db/dcp_sharded_upgrade_test.go

```diff
@@ -0,0 +1,283 @@
+      "name": "db0x2d9928b7_index_438836eb24955a74_120a6de6",
+      "uuid": "3ccf34ea23fb9727",
+      "indexType": "syncGateway-import-",
+      "indexName": "db0x2d9928b7_index",
+      "indexUUID": "438836eb24955a74",
+      "sourceType": "couchbase-dcp-sg",
+      "sourceName": "%[1]s",
+      "sourceUUID": "%[2]s",
+      "sourcePartitions": "$SourcePartitions",
+      "nodes": {
+        "680cce1d31e72b69": {
+          "canRead": true,
+          "canWrite": true,
+          "priority": 0
+        }
+      },
+      "indexParams": {
+        "destKey": "%[1]s_import"
+      },
+      "sourceParams": {
+        "includeXAttrs": true,
+        "sg_dbname": "%[1]s"
+      }
+    },
+    "db0x2d9928b7_index_438836eb24955a74_15a93994": {
+      "name": "db0x2d9928b7_index_438836eb24955a74_15a93994",
+      "uuid": "4c167ac08c668923",
+      "indexType": "syncGateway-import-",
+      "indexName": "db0x2d9928b7_index",
+      "indexUUID": "438836eb24955a74",
+      "sourceType": "couchbase-dcp-sg",
+      "sourceName": "%[1]s",
+      "sourceUUID": "%[2]s",
+      "sourcePartitions": "$SourcePartitions",
+      "nodes": {
+        "680cce1d31e72b69": {
+          "canRead": true,
+          "canWrite": true,
+          "priority": 0
+        }
+      },
+      "indexParams": {
+        "destKey": "%[1]s_import"
+      },
+      "sourceParams": {
+        "includeXAttrs": true,
+        "sg_dbname": "%[1]s"
+      }
+    }
+  },
+  "implVersion": "5.5.0",
+  "warnings": {
+    "db0x2d9928b7_index": []
+  }
+}`
+
+func TestShardedDCPUpgradeHelium(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Requires Couchbase Server")
+	}
+	if !base.IsEnterpriseEdition() {
+		t.Skip("EE-only test")
+	}
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyDCP, base.KeyImport, base.KeyCluster)
+	tb := base.GetTestBucket(t)
+	defer tb.Close()
+	bucketUUID, err := tb.UUID()
+	require.NoError(t, err, "get bucket UUID")
+
+	numVBuckets, err := tb.GetMaxVbno()
+	require.NoError(t, err)
+
+	const numPartitions = 4
+
+	require.NoError(t, tb.SetRaw(base.SyncPrefix+"cfgindexDefs", 0, nil, []byte(fmt.Sprintf(indexDefs, tb.GetName(), bucketUUID))))
+	require.NoError(t, tb.SetRaw(base.SyncPrefix+"cfgnodeDefs-known", 0, nil, []byte(nodeDefs)))
+	require.NoError(t, tb.SetRaw(base.SyncPrefix+"cfgnodeDefs-wanted", 0, nil, []byte(nodeDefs)))
+	planPIndexesJSON := preparePlanPIndexesJSON(t, tb, numVBuckets, numPartitions)
+	require.NoError(t, tb.SetRaw(base.SyncPrefix+"cfgplanPIndexes", 0, nil, []byte(planPIndexesJSON)))
+
+	// Write a doc before starting the dbContext to check that import works
+	const (
+		testDoc1 = "testDoc1"
+		testDoc2 = "testDoc2"
+	)
+	require.NoError(t, tb.SetRaw(testDoc1, 0, nil, []byte(`{}`)))
+
+	t.Log("TEST: setup complete")
+
+	// Start this retry loop asynchronously because the change may happen before NewDatabaseContext returns
```

Comment:
Is this comment still accurate?  I don't see what retry loop is being run asynchronously here.